### PR TITLE
Detache strategy

### PR DESCRIPTION
**Pull request by BioPhoton**

### CODE REVIEW

File: libs/template/docs/render-strategies.md [hunk 1]
```diff
@@ -41,62 +41,84 @@ export class PerformanceAwareComponent {
...
+The detach strategy detaches the component from Angulars change detection.
+With every new value it re-attaches the component/embedded view to the change detection,
+renders the new value and detaches again.
 
-As detectChanges is synchronous and has no built-in coalescing of rendering
-like `ChangeDetectorRef#markForCheck` or `ɵmarkDirty` have, we have to apply our own coalescing.
-It is also _scoped_ on the component level. (see [Concepts](https://github.com/BioPhoton/rx-angular/tree/master/libs/template/docs/concepts.md) for more information)
+If a component is detached the input bindings will still receive values.
+Also the internal logic will work as expected including the use of `ViewChild`.
+Only the template will not be updated.
```
Comment (hoebbelsB) [inline libs/template/docs/render-strategies.md:96]:
```suggestion
Additional information about `detached` views:

* If a view is detached, its input bindings will still receive values
* Also any related code will get executed properly as well as using Angulars internal APIs such as `ViewChild`
* Detached views are not getting checked when their parent components call `detectChanges`
* `HostBindings` like `[class]` `@animation` ... are not getting updated properly, when the parent component does not detect a new change
```

**Review by hoebbelsB — CHANGES_REQUESTED** [review]
THX for updating the docs! I have reviewed most of it and got some suggestions!

File: libs/template/src/lib/render-strategies/strategies/detach-strategy.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    config.cdRef.reattach();
+    config.cdRef.detectChanges();
+    config.cdRef.detach();
+  };
+  const behavior = (o) => o.pipe(
+          coalesceWith(promiseDurationSelector, component),
+          // dispose previous render attempts (tick schedules it into the future)
+          switchMap((v) => tick.pipe(map(() => v))),
+          tap(renderMethod)
+    );
```
[the remark this answers is not in the document]
Comment (BioPhoton) [reply libs/template/src/lib/render-strategies/strategies/detach-strategy.ts:68]: removed as it was useless for this manner

**Review by hoebbelsB — APPROVED** [review]
:1st_place_medal:
